### ISSUE TITLE
Suggestion: `derive` Cargo feature

### ISSUE THREAD

**Issue by RReverser** [Marwes/serde_state#5]
Nowadays, in `serde` the recommended way to import derives is to use `serde = { version = ..., features = ["derive"] }` in Cargo.toml. It takes care of reexporting the macros, to instead of writing

```
use serde::{Serialize, Deserialize};
use serde_derive::{Serialize, Deserialize};
```

to import both traits and derives you can just write

```
use serde::{Serialize, Deserialize};
```

Would it be possible to make `serde_state` have the same feature so that derives would be automatically included under corresponding trait names?

**Comment by Marwes** [comment]
This may be fixed if we do another merge from `serde` upstream, though that can be quite annoying at times.

**Comment by RReverser** [comment]
@Marwes Is there any way to detach them / reuse serde bits via public APIs instead?

**Comment by Marwes** [comment]
No, this is basically `serde_derive` but modified to thread a `state` argument through. `serde_derive` itself does not (and could not, without seriously invasive changes) expose an API for this.

**Comment by RReverser** [comment]
> `serde_derive` itself does not (and could not, without seriously invasive changes) expose an API for this.

They do have [`serde_derive_internals`](https://crates.io/crates/serde_derive_internals) - that API is not stable either, but maybe sufficient? Just thinking of ways to make upgrades a bit easier.
